clients/lsp-tailwindcss.el

```diff
@@ -0,0 +1,212 @@
+  @variants directive(https://tailwindcss.com/docs/functions-and-directives/#variants)"
+  :type '(choice
+          (const "ignore")
+          (const "warning")
+          (const "error"))
+  :package-version '(lsp-mode . "7.1"))
+
+(defcustom lsp-tailwindcss-lint-invalid-config-path "error"
+  "Unknown or invalid path used with the
+  theme helper(https://tailwindcss.com/docs/functions-and-directives/#theme)"
+  :type '(choice
+          (const "ignore")
+          (const "warning")
+          (const "error"))
+  :package-version '(lsp-mode . "7.1"))
+
+(defcustom lsp-tailwindcss-lint-invalid-tailwind-directive "error"
+  "Unknown value used with the
+  @tailwind directive(https://tailwindcss.com/docs/functions-and-directives/#tailwind)"
+  :type '(choice
+          (const "ignore")
+          (const "warning")
+          (const "error"))
+  :package-version '(lsp-mode . "7.1"))
+
+(defcustom lsp-tailwindcss-experimental-class-regex '()
+  ""
+  :type '(repeat string)
+  :package-version '(lsp-mode . "7.1"))
+
+(defcustom lsp-tailwindcss-experimental-show-pixel-values '()
+  ""
+  :type '(repeat string)
+  :package-version '(lsp-mode . "7.1"))
+
+(lsp-register-custom-settings
+ '(("tailwindCSS.includeLanguages" lsp-tailwindcss-include-languages)
+   ("tailwindCSS.emmetCompletions" lsp-tailwindcss-emmet-completions)
+   ("tailwindCSS.colorDecorators" lsp-tailwindcss-color-decorators)
+   ("tailwindCSS.validate" lsp-tailwindcss-validate)
+   ("tailwindCSS.lint.cssConflict" lsp-tailwindcss-lint-css-conflict)
+   ("tailwindCSS.lint.invalidApply" lsp-tailwindcss-lint-invalid-apply)
+   ("tailwindCSS.lint.invalidScreen" lsp-tailwindcss-lint-invalid-screen)
+   ("tailwindCSS.lint.invalidVariant" lsp-tailwindcss-lint-invalid-variant)
+   ("tailwindCSS.lint.invalidConfigPath" lsp-tailwindcss-lint-invalid-config-path)
+   ("tailwindCSS.lint.invalidTailwindDirective" lsp-tailwindcss-lint-invalid-tailwind-directive)
+   ("tailwindCSS.experimental.classRegex" lsp-tailwindcss-experimental-class-regex)
+   ("tailwindCSS.experimental.showPixelValues" lsp-tailwindcss-experimental-show-pixel-values)))
+
+(defun lsp-tailwindcss--config-updated ()
+  ;; FIXME: Update decorations in editor, see original implementation
+  ;; https://github.com/tailwindlabs/tailwindcss-intellisense/blob/b3c17cf5c9d1d74091a4a5ee0504696bb2ce1c46/packages/tailwindcss-intellisense/src/lib/registerColorDecorator.ts#L125
+  t)
+
+(defun lsp-tailwindcss--config-error ()
+  ;; FIXME: Update decorations in editor and show error message, see original implementation
+  ;; https://github.com/tailwindlabs/tailwindcss-intellisense/blob/b3c17cf5c9d1d74091a4a5ee0504696bb2ce1c46/packages/tailwindcss-intellisense/src/lib/registerColorDecorator.ts#L129
+  ;; https://github.com/tailwindlabs/tailwindcss-intellisense/blob/b3c17cf5c9d1d74091a4a5ee0504696bb2ce1c46/packages/tailwindcss-intellisense/src/lib/registerConfigErrorHandler.ts#L5
+  t)
+
+(defun lsp-tailwindcss--get-configuration ()
```

Comment:
this one you will have to return a map with tab size in it: `(symbol-value (lsp--get-indent-width major-mode))`

Reply:
The original implementation seems to be get all configuration related to tailwindcss by calling `...Workspace.getConfiguration('tailwindCSS', scope)`.
Is there any function in lsp-mode equivalent to that of vscode?

Reply:
`lsp-configuration-section`